tests/test_socket_server.py

```diff
@@ -33,94 +30,28 @@
-        await self.server.start_task
-        self.reader, self.writer = await asyncio.open_connection(
-            host=tcpip.LOCAL_HOST, port=self.server.port
-        )
-        # Give time to the socket server to respond.
-        await self.connected_future
-        assert self.server.connected
-        self.log.info("===== End of asyncSetUp =====")
-
-    async def asyncTearDown(self) -> None:
-        self.log.info("===== Start of asyncTearDown =====")
-        if self.server.connected:
-            await self.server.disconnect()
-        if self.writer:
-            self.writer.close()
-            await self.writer.wait_closed()
-        await self.server.exit()
-
-    async def read(self) -> dict[str, Any]:
-        """Read a string from the reader and unmarshal it
-
-        Returns
-        -------
-        data : `dict`
-            A dictionary with objects representing the string read.
-        """
-        read_bytes = await asyncio.wait_for(
-            self.reader.readuntil(tcpip.TERMINATOR), timeout=TIMEOUT
-        )
-        data = json.loads(read_bytes.decode())
-        return data
-
-    async def write(self, **data: dict[str, Any]) -> None:
-        """Write the data appended with a tcpip.TERMINATOR string.
-
-        Parameters
-        ----------
-        data:
-            The data to write.
-        """
-        st = json.dumps({**data})
-        assert self.writer is not None
-        self.writer.write(st.encode() + tcpip.TERMINATOR)
-        await self.writer.drain()
-
-    async def connect_callback(self, server: common.SocketServer) -> None:
-        if not self.connected_future.done():
-            self.connected_future.set_result(server.connected)
-            await self.server.connect_callback(server)
-
-    async def test_disconnect(self) -> None:
-        self.connected_future = asyncio.Future()
-        assert self.server.connected
-        await self.assert_configure(name="TEST_DISCONNECT")
-        await self.write(command=common.Command.DISCONNECT, parameters={})
-        # Give time to the socket server to clean up internal state and exit.
-        await self.connected_future
-        assert not self.server.connected
-
-    async def test_exit(self) -> None:
-        self.connected_future = asyncio.Future()
-        assert self.server.connected
-        await self.assert_configure(name="TEST_EXIT")
-        await self.write(command=common.Command.EXIT, parameters={})
-        # Give time to the socket server to clean up internal state and exit.
-        await self.connected_future
-        assert not self.server.connected
+        ) as server:
+            command_handler = common.MockCommandHandler(
+                callback=server.write_json, simulation_mode=1
+            )
+            server.set_command_handler(command_handler)
+            yield server
 
     async def assert_configure(
         self,
+        client: tcpip.Client,
```

Comment:
Please add a doc string for this method explaining what it does and (preferably) documenting the parameters.

Reply:
Thanks for pushing back on this. It turned out that several parameters weren't used anymore so I cleaned up the code and added the doc string.